srcs/ray.c

```diff
@@ -6,31 +6,67 @@
 /*   By: tgrivel <marvin@42lausanne.ch>             +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2022/08/24 17:04:24 by tgrivel           #+#    #+#             */
-/*   Updated: 2022/09/15 10:43:20 by tgrivel          ###   ########.fr       */
+/*   Updated: 2022/09/20 05:53:39 by tgrivel          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include	"cub3d.h"
 
-// ray[X, Y]: The position of the ray
-static void	ray(t_info *info, float delta[2])
+// Collect the distance to the next vertical line and the next horizontal line
+static void
+	ray_distance(t_info *info, float distance[2],
+		float hit[2][2], float delta[2])
 {
-	float	ray[2];
+	distance[X] = -1.0f;
+	distance[Y] = -1.0f;
+	if (0.0f < delta[X])
+		distance[X] = vertical_right(info, hit[X], delta[X], delta[Y]);
+	if (delta[X] < 0.0f)
+		distance[X] = vertical_left(info, hit[X], delta[X], delta[Y]);
+	if (0.0f < delta[Y])
+		distance[Y] = horizontal_down(info, hit[Y], delta[X], delta[Y]);
+	if (delta[Y] < 0.0f)
+		distance[Y] = horizontal_up(info, hit[Y], delta[X], delta[Y]);
```

Comment:
See previous suggestion
```suggestion
	if (0.0f < delta[X])
		distance[X] = vertical_right(info, hit[X], delta);
	if (delta[X] < 0.0f)
		distance[X] = vertical_left(info, hit[X], delta);
	if (0.0f < delta[Y])
		distance[Y] = horizontal_down(info, hit[Y], delta);
	if (delta[Y] < 0.0f)
		distance[Y] = horizontal_up(info, hit[Y], delta);
```

Reply:
Done: 41746eafbc9bf3ef2ac7237d6dec1655f223ef24